### DISCUSSION TITLE
How about an installer ?

### DISCUSSION THREAD

**Discussion by NorHei** [Milkdown/milkdown#1231] [General]
I looked for alternatives to Typora as the developers are a bit "unflexible" when it comes to extensions of their product...

So i stumbled over Milkdown as i really liked the idea of having just a basic editor and everything else is just modules... The first thing that completely dissappointed me was that there was no easy way to intall and test it.

I would guess most pepole using Typora, aren't into building node apps by themselfes and most havent even an idea on how to get node.js up and running. So as soon as they arrive here they find no installer , and instantly turn away. And to be true, thats a pitty...

Basically i am a backend PHP developer and have some experience whith Python but almost never had much to do whith Node.  
Ok i set up micromamba for a new enviroment and installed node into it. Next question was what package do i need ? Ok, lets try....

`npm i milkdown`
npm ERR! No versions available for milkdown

A quick search on  https://www.npmjs.com gave me milkdown/core as an package. 

This failed miserably as i forgot the @ in: `npm i milkdown/core` :-)

The next showtopper that got me was my silly old github account, as i usually use a locally hosted version of gitlab. But for downloading the repo it asked me to authenticate whith github and github did tell me, "no that features whithout 2FA" ... so i had to have angit account and  setup 2FA with git first, before anything would happen ;-) 

Ok now it installed ... but 

Created a editor.js whith the content from first example in getting started 

`node .\editor.js`
And nothing happens ....

Thats the point where i gave up and just thought you schould know about. 
The "getting started" is a  "go away if you are no node developer". 

I really like the idea, please update the getting started.

**Comment by NorHei** [comment]
Ok got it running ...

Personally i prefer to put everything in a Virtual enviroment so is dosent tamper whith the rest of my system. 
Basically i use micromamba , but you can use conda or something simmilar as well(even Docker if you like). 

After Installing Micromamba:
https://mamba.readthedocs.io/en/latest/installation/micromamba-installation.html

i created a folder to contain all this (\milktest) and cd into the folder. 

Be sure to have mircomamba updated
`micromamba self-update`

Append the comunity channel 
`micromamba config append channels conda-forge `

Create the Virtual enviroment and activate it
`micromamba create -n milktest `
`micromamba activate milktest`

Install Node and update to recent stable 
`micromamba install -c conda-forge nodejs `
`npm install -g node@20.11.0 `
`npm install -g npm@latest `

Fetch the example Repo(Git schould be installed): 
`git clone https://github.com/Milkdown/examples.git `

Dive into an example (e.G. cd vanilla-commonmark)

Install and start the project:
`npm install `
`npm start`

I Installed pnpm via npm but it failed installing when diving into the example/subdirs :
```
pnpm install
pnpm: The term 'pnpm' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
```
But basic npm did the job, so i dont care.

The frustrating thing now, is that there is no example of an at least halfway functional editor that can load and save files and is at least half way usable. If there is a usable Version out there maybe simply link it ? 

This project has an enormous potential but it seems you are missing the basics. Get a Product running with basic functionality and the advertise it. Obsidian started as a simple notetaking app, now its a big conmpany.       
People expect either a functional webapp like draw.io or alternaternavely a simple website integration via a single file loaded from a CDN. Or at bare minimum a halfway functional/installable  example app. 

You can call me a grumpy old man but i get angry when a project whith a potential like this fucks up the basics....

**Comment by Saul-Mirone** [comment]
An installer is too far for this project. But I do notice some difficulties for beginners to start with milkdown. I'm working on a package called [crepe](https://milkdown-storybook.vercel.app/?path=/story/crepe-main--empty) to provide an editor that can work just out of box without much config and code. And it will also be a good example :)

**Reply by quank123wip** [reply]
Very happy to see something like this. I think we should do something to gain more popularity as we might need more active maintainer for this project. I tried to suggest my friends to try milkdown and use it as their daily drive, but in the end most of them gave up. They suggested that the default instance provided in milkdown.dev has a nice UI, but the UX still needs lots improvements and some new features, and the PWA can't fits them compared with stackedit. My idea is that we should provide better examples and more presets for users to choose, not only one nord theme presets. It requires much hard work and I'm not posting this as a request to you. I'll work on this months later, make some new themes, presets and new plugins. I hope I can make some of them as official package included in milkdown. After that, I think I would try to make a lightweight tauri app for users want to replace Typora or Marktext. It's time for some community power now.
